analyses/survival-analysis/survival-analysis_immune.Rmd

```diff
@@ -0,0 +1,324 @@
+
+> No cell types are meaningfully signficant.
+
+```{r cox_cell_type}
+
+# Define the terms for a quantiseq model. These are used several times in this notebook
+quantiseq_terms <- "B_cell + Macrophage_M1 + Macrophage_M2 + Monocyte + Myeloid_dendritic_cell + Neutrophil + NK_cell + T_cell_CD4_non_regulatory + T_cell_CD8 + T_cell_regulatory_Tregs" 
+
+
+# saving so P-values can be programmatically grabbed
+fitted <- fit_save_model(combined_df_stranded,
+                         quantiseq_terms,
+                         file.path(results_dir, "cox_terms_quantiseq.RDS"),
+                         model_type = "multivariate")
+fitted
+```
+
+Above, no effects are significant at P<0.01. Only one cell type, `T_cell_CD8`, is marginally significant ($P=`r fitted$p.value[fitted$term == "T_cell_CD8"]`$), but it's confidence interval is *huge* so the exact effect is challenging to discern.
+
+## Analysis: quantiseq cell type fractions and PDL-1  across entire cohort
+
+> No effects are significant
+
+```{r cox_cell_type_cd274}
+fit_save_model(combined_df_stranded,
+               paste0(quantiseq_terms, "+CD274"),
+               file.path(results_dir, "cox_terms_quantiseq.RDS"),
+               model_type = "multivariate")
+```
+
+
+## Analysis: quantiseq cell type fractions and PDL-1 for HGG 
+
+> No effects are significant
+
+```{r cox_hgg_cell_type_cd274}
+hgg_groups <- c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma")
+hgg_df <- combined_df_stranded %>%
+  filter(cancer_group %in% hgg_groups)
+
+fit_save_model(combined_df_stranded,
+               paste0(quantiseq_terms, "+CD274"),
+               file.path(results_dir, "cox_hgg_terms_quantiseq.RDS"),
+               model_type = "multivariate")
+```
+
+
+## Determine suitable cancer groups for analysis
+
+For cancer group analyses, we only want to consider groups with at least THREE DECEASED, again considering stranded only. We find those groups here:
+```{r cancer_groups}
+suitable_cancer_groups <- combined_df_stranded %>%
+  drop_na() %>%
+  count(cancer_group,OS_status) %>%
+  spread(OS_status, n,
+         # When there are no events, record the event count as 0 instead of the default NA
+         fill = 0) %>%
+  filter(DECEASED >= 3) 
+
+# show result: there are 5 groups to keep
+suitable_cancer_groups
+```
+
+For molecular signatures analysis, we want to only consider cancer groups, of those identified above, that have more than 1 molecular subtype.
+
+```{r suitable_subtypes}
+combined_df_stranded %>%
+  filter(cancer_group %in% suitable_cancer_groups$cancer_group) %>%
+  select(cancer_group, molecular_subtype) %>%
+  distinct() %>%
+  # count number of subtypes
+  count(cancer_group) %>%
+  filter(n > 1) -> suitable_subtypes
+
+
+# n is how many subtypes the group has. There needs to be more than 1.
+suitable_subtypes
+```
+
+
+
+## Analysis: quantiseq cell type fractions and PDL-1 for each cancer group of interest
```

Comment:
Can you check the forest plot output files for this? They seem out of date. For ex, in the DMG plot, CD274 is not significant, but in the notebook, it has a p = 0.04. Can you print out new forest plots, just to be sure everything is running as it should be (eg the really easy QC was looking at the Ns on the plots)

Reply:
I forgot to remove old forest plot files so I will remove them.